### PR TITLE
add an error count field to the Flutter.Error event

### PR DESCRIPTION
**Pull request by devoncarew**
## Description

- add an error count field to the Flutter.Error event; this will be used by clients to display errors after an initial error differently (with reduced emphasis, or collapsed by default)

## Related Issues

none

## Tests

I added the following tests:

- `widgets/widget_inspector_test.dart` : `ext.flutter.inspector.structuredErrors`

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by devoncarew** [comment]
cc @DanTup, this will add a `errorsSinceReload` field to the json received for the `Flutter.Error` event. We can use that info to hide (or summarize by default) errors after an initial error. The error count automatically resets after a hot reload or hot restart; the first error sent will have a value of `0` for the field.

**Comment by DanTup** [comment]
Is it always correct to hide errors if count > 0? (for ex., if an error was only in a sub-widget, could a user interact with the app and generate other errors they might expect to see logged?)

**Comment by devoncarew** [comment]
We  may instead want to aggressively summarize the error. You can see this in the 2nd example from the flutter_error_studies app - there's a 2nd error generated. The standard text error output just summarizes with a single text file. We may want to do that, or a few very terse lines.

Note that `errorsSinceReload` will not be attached to errors from older versions of flutter.

**Comment by DanTup** [comment]
> We may instead want to aggressively summarize the error.

That sounds reasonable, I'll play around and see how it looks.

> Note that `errorsSinceReload` will not be attached to errors from older versions of flutter.

Noted, thanks!

File: packages/flutter/test/widgets/widget_inspector_test.dart [hunk 1]
```diff
@@ -2331,6 +2331,41 @@ class TestWidgetInspectorService extends Object with WidgetInspectorService {
...
+      expect(flutterErrorEvents, hasLength(1));
+
+      // validate the error contents
+      final Map<Object, Object> error = flutterErrorEvents.first;
+      expect(error['description'], 'Exception caught by rendering library');
+      expect(error['children'], isEmpty);
+
+      // validate that we received an error count
+      expect(error['errorsSinceReload'], 0);
+    });
```
Comment (jacob314) [inline packages/flutter/test/widgets/widget_inspector_test.dart]: Thanks for adding a test for this! Can you also dispatch a second error (or the same error) and verify that the errorsSinceReload count increases?
You could also dispatch the event for a hot reload and verify the count is back to zero.

Comment (devoncarew) [reply packages/flutter/test/widgets/widget_inspector_test.dart]: 👍 will do

**Review by devoncarew — COMMENTED** [review]
ptal - updated with two additional tests of the error count

File: packages/flutter/test/widgets/widget_inspector_test.dart [hunk 2]
```diff
@@ -2349,21 +2349,51 @@ class TestWidgetInspectorService extends Object with WidgetInspectorService {
...
+        expect(error['errorsSinceReload'], 0);
+
+        // send a second error
+        FlutterError.reportError(FlutterErrorDetailsForRendering(
+          library: 'rendering library',
+          context: ErrorDescription('also during layout'),
+          exception: StackTrace.current,
+        ));
+
+        // validate that the error count increased
```
Comment (jacob314) [inline packages/flutter/test/widgets/widget_inspector_test.dart]: Nit: make all these comments start with a capital letter and end with a period to match the rest of the file.

**Review by jacob314 — APPROVED** [review]
lgtm

File: packages/flutter/test/widgets/widget_inspector_test.dart (hunk 2 above)
Comment (devoncarew) [reply packages/flutter/test/widgets/widget_inspector_test.dart]: done

**Pull request merged**